backend/control/control.go

```diff
@@ -169,13 +120,130 @@ func TriggerScan(postgresDb *gorm.DB) error {
+}
+
+func CartographyExecutionLoop(postgresDb *gorm.DB, driver neo4j.Driver, ruleDataList []models.RuleData,
+	rulesToExecute []types.Rule) {
+	for {
+		accountReadyToBeScanned, err := GetAccountReadyToBeScanned(postgresDb)
+		if err != nil {
+			log.Printf("Error in fetching the details of account with status as READY: %v", err)
+			continue
+		} else if accountReadyToBeScanned == (models.AccountDetails{}) {
+			// No accounts found which are ready to be scanned
+			continue
+		}
+
+		if err := StartCartographyJob(accountReadyToBeScanned); err != nil {
+			log.Printf("Error in starting cartography job for Account Name : %v \nError: %v", accountReadyToBeScanned.AccountName, err)
+		}
+
+		MonitorCartographyJobStatusLoop(postgresDb, accountReadyToBeScanned.AccountName)
+
+		// Change CARTOGRAPHY_PASSED status to RULES_RUNNNING
+		if tx := postgresDb.Model(&models.AccountDetails{}).Where("scan_status = ?", "CARTOGRAPHY_PASSED").Updates(map[string]interface{}{"scan_status": "RULES_RUNNING"}); tx.Error != nil {
+			log.Printf("Error in updating the scan status of CARTOGRAPHY_PASSED accounts to RULES_RUNNNING: %v", tx.Error)
+			continue
+		}
+
+		if areAccountsReadyForExecutingRules, err := CheckAccountsReadyForExecutingRules(postgresDb); err != nil {
+			log.Printf("Error in checking RULES_RUNNING status of accounts")
+			continue
+		} else if areAccountsReadyForExecutingRules {
+			executeRulesErr := ExecuteRules(postgresDb, driver, ruleDataList, rulesToExecute)
+			if executeRulesErr != nil {
+				log.Printf("Error in executing rules: %v", executeRulesErr)
+				continue
+			}
+		}
+	}
+}
+
+func MonitorCartographyJobStatusLoop(postgresDb *gorm.DB, accountName string) {
+	var currScanStatus = "READY"
+	for {
+		time.Sleep(time.Second * 10)
+		sr, err := GetScanStatus()
+		if err != nil {
+			log.Printf("Error fetching cartography job status: %v", err)
+			continue
+		}
+		if sr.Status != currScanStatus {
+			tx := postgresDb.Model(&models.AccountDetails{}).Where("account_name = ?", accountName).Updates(map[string]interface{}{"scan_status": sr.Status})
+			if tx.Error != nil {
+				log.Printf("Error updating the scan status of account: %v", accountName)
+				continue
+			}
+			currScanStatus = sr.Status
+		}
+		if sr.Status == "CARTOGRAPHY_PASSED" || sr.Status == "FAILED" {
+			break
+		}
+		tx := postgresDb.Model(&models.AccountDetails{}).Where("account_name = ?", accountName).Updates(map[string]interface{}{"running_time": sr.RunningTime})
+		if tx.Error != nil {
+			continue
+		}
+	}
+	log.Printf("Cartography job scanning completed for Account name: %v", accountName)
+
+	if currScanStatus == "FAILED" {
+		if tx := postgresDb.Model(&models.AccountDetails{}).Where("account_name = ?", accountName).Update("last_scan_completed", time.Now()); tx.Error != nil {
+			log.Printf("Error in updating the last_scan_completed of account %v", accountName)
+		}
+	}
+}
+
+func GetAccountReadyToBeScanned(postgresDb *gorm.DB) (models.AccountDetails, error) {
```

Comment:
also document what this function does, when it returns an error and when nil

Reply:
Can you tell why the comments are necessary, I think we should avoid comments as much as possible and try to incorporate understanding through more descriptive names, as we tend to not change comments when we make some changes in the code in future which will ultimately lead to clutter and confusion

Reply:
Makes sense - I think you can leave out comments for this function.

The higher level functions like ExecuteRules and CartographyExecutionLoop should have them, since they do many things and it may not be obvious for the names and scanning through the code at first glance

Reply:
Okay